RealmSwift/Object.swift

```diff
@@ -542,13 +542,13 @@ public final class DynamicObject: Object {
         get {
             let value = RLMDynamicGetByName(self, key).flatMap(coerceToNil)
             if let array = value as? RLMArray<AnyObject> {
-                return List<DynamicObject>(objc: array)
+                return list(from: array)
             }
             if let set = value as? RLMSet<AnyObject> {
-                return MutableSet<DynamicObject>(objc: set)
+                return mutableSet(from: set)
             }
             if let dictionary = value as? RLMDictionary<AnyObject, AnyObject> {
-                return Map<String, DynamicObject>(objc: dictionary)
```

Comment:
This is an issue, Map only supports optional objects. So changing this here and on `dynamicMap` will be a breaking change.